### PR TITLE
update fs with get and count methods

### PR DESCRIPTION
**Pull request by derric-d**
add method to retrieve count of objects by class and a method to retrieve specific object by class and id

### CODE REVIEW

File: models/engine/file_storage.py [hunk 1]
```diff
@@ -68,3 +68,12 @@ def delete(self, obj=None):
     def close(self):
         """call reload() method for deserializing the JSON file to objects"""
         self.reload()
+
+    def get(self, cls, id):
+        """get the obejects by id, using dict's get()"""
```
Comment (allelomorph) [inline models/engine/file_storage.py]: Typo in "objects," and feels a bit vague to me. I won't be too pedantic about docstrings, and want to compromise. But you will find that mine tend to be a bit more verbose, and follow the Google style guide that they gave us a while back:
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: models/engine/file_storage.py [hunk 2]
```diff
@@ -68,3 +68,12 @@ def delete(self, obj=None):
...
+        if cls in classes and id and type(id) == str:
```
Comment (allelomorph) [inline models/engine/file_storage.py]: `if <var> in <dict>` checks its keys, but if we are to match the use of `test_get_count.py` and not the task description, we want to check `if cls in classes.values()`

File: models/engine/file_storage.py [hunk 3]
```diff
@@ -68,3 +68,12 @@ def delete(self, obj=None):
...
+            return self.__objects.get(cls + '.' + id, None)
```
Comment (allelomorph) [inline models/engine/file_storage.py]: Similarly, if we are matching the test program and being fed a `cls` that is a `State` object and not the string `'State'`, then we need to change this to `cls.__name__ + '.' + id`.

Also, I like the readability of `+` concatenation, but have been converted to the gospel of `join()`. :)

**Review by allelomorph — CHANGES_REQUESTED** [review]
Derric -

Thanks for getting the ball rolling on writing the actual production code for this project! I have few changes to request after debugging this locally, they can be found in the line by line comments.

Most of the code issues arise with a contradiction in the task description: The task instructions say that the method parameter `cls` should be the "class name," which implies a string, or (using State objects as an example) `State.__name__`  -> `'State'`. But the `test_get_count.py` code feeds `get()` the `State` class, not its name string. I have submitted the question to Guillaume on the improvement channel.

For now, I am working on the assumption that it is more important to match the test program, since it may indicate how the checker works. So all comments are based on that for now.

Thanks again,

-Sam

File: models/engine/file_storage.py (hunk 1 above)
Comment (derric-d) [reply models/engine/file_storage.py]: do we change the other method comments?

**Pull request merged**